src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs

```diff
@@ -0,0 +1,216 @@
+using Standard;
+using System.Windows.Appearance;
+using System.Windows.Media;
+using Microsoft.Win32;
+using System.Collections;
+using System.Collections.Generic;
+using System.Windows.Interop;
+
+namespace System.Windows;
+
+internal static class ThemeManager
+{
+
+    #region Constructor
+
+    static ThemeManager()
+    {
+        _cachedThemeDictionaryUris = new List<Uri>();
+
+        if (Application.Current != null)
+        {
+            foreach (ResourceDictionary mergedDictionary in Application.Current.Resources.MergedDictionaries)
```

Comment:
Check for NRE case

Reply:
Not needed.
Application.Resources is a property where if resources is null, it returns a new instance of ResourceDictionary.

Similary, in ResourceDictionary, we have a property MergedDictionary, which does the same thing, if the member is null, we create a new instance and return that.